lib/chef/knife/supermarket_list.rb

```diff
@@ -37,25 +37,40 @@ class SupermarketList < Knife
         default: "https://supermarket.chef.io",
         proc: Proc.new { |supermarket| Chef::Config[:knife][:supermarket_site] = supermarket }
 
+      option :sort_by,
+        long: "--sort-by SORT",
+        description: "Use to sort the records",
+        in: %w{recently_updated recently_added most_downloaded most_followed}
+
+      option :owned_by,
+        short: "-o USER",
+        long: "--owned-by USER",
+        description: "Show cookbooks that are owned by the USER"
+
       def run
+        url_params = []
+        url_params << "&order=#{config[:sort_by]}" if config[:sort_by]
+        url_params << "&user=#{config[:owned_by]}" if config[:owned_by]
+
+        query = url_params.join unless url_params.empty?
+        cookbooks_list = get_cookbook_list(query)
         if config[:with_uri]
-          cookbooks = {}
-          get_cookbook_list.each { |k, v| cookbooks[k] = v["cookbook"] }
-          ui.output(format_for_display(cookbooks))
+          ui.output(format_for_display(cookbooks_list))
         else
-          ui.msg(ui.list(get_cookbook_list.keys.sort, :columns_down))
+          ui.msg(ui.list(cookbooks_list.keys, :columns_down))
         end
       end
 
-      def get_cookbook_list(items = 10, start = 0, cookbook_collection = {})
+      def get_cookbook_list(query, items = 1000, start = 0, cookbook_collection = {})
         cookbooks_url = "#{config[:supermarket_site]}/api/v1/cookbooks?items=#{items}&start=#{start}"
+        cookbooks_url << query if query
```

Comment:
Could we reduce the number of things changing in this by moving the `sort_by` and `owned_by` parameter additions to here?

```
cookbooks_url = "#{config[:supermarket_site]}/api/v1/cookbooks?items=#{items}&start=#{start}"
cookbooks_url << "&order=#{config[:sort_by]}" if config[:sort_by]
cookbooks_url << "&user=#{config[:owned_by]}" if config[:owned_by]
```

I think that would leave the `run` method untouched and the `get_cookbook_list` method solely responsible for URL string construction.

Reply:
Ohh Yup, Thanks